src/hotspot/cpu/aarch64/aarch64_vector_ad.m4

```diff
@@ -3616,6 +3616,47 @@ instruct vmaskcmp_sve(pReg dst, vReg src1, vReg src2, immI cond, rFlagsReg cr) %
   %}
   ins_pipe(pipe_slow);
 %}
+dnl
+dnl VMASKCMP_SVE_IMM_I($1      , $2            )
+dnl VMASKCMP_SVE_IMM_I(type_imm, type_condition)
+define(`VMASKCMP_SVE_IMM_I', `
+instruct vmask$2_immI_sve(pReg dst, vReg src, $1 imm, immI_$2_cond cond, rFlagsReg cr) %{
+  predicate(UseSVE > 0);
+  match(Set dst (VectorMaskCmp (Binary src (ReplicateB imm)) cond));
```

Comment:
The ReplicateXNodes used in match rules are also different in these two marcos.
I think we needn't merge these two marcos since this will introduce some if-else statements which will reduce the readability.

Reply:
> @theRealAph The ReplicateXNodes used in match rules are also different in these two marcos. I think we needn't to merge these two marcos since this will introduce some if-else statements which will reduce the readability.

It won't introduce if-else, surely. Not if you make the parts that are different into parameters. Then a reviewer cand immediately see which parts of the macros are actually different, rather than needing to do a deep study.;